client/rest-high-level/src/main/java/org/elasticsearch/client/indices/CreateIndexRequest.java

```diff
@@ -0,0 +1,394 @@
+     */
+    public CreateIndexRequest aliases(XContentBuilder source) {
+        return aliases(BytesReference.bytes(source), source.contentType());
+    }
+
+    /**
+     * Sets the aliases that will be associated with the index when it gets created
+     */
+    public CreateIndexRequest aliases(String source, XContentType contentType) {
+        return aliases(new BytesArray(source), contentType);
+    }
+
+    /**
+     * Sets the aliases that will be associated with the index when it gets created
+     */
+    public CreateIndexRequest aliases(BytesReference source, XContentType contentType) {
+        // EMPTY is safe here because we never call namedObject
+        try (XContentParser parser = XContentHelper.createParser(NamedXContentRegistry.EMPTY,
+                DeprecationHandler.THROW_UNSUPPORTED_OPERATION, source, contentType)) {
+            //move to the first alias
+            parser.nextToken();
+            while ((parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                alias(Alias.fromXContent(parser));
+            }
+            return this;
+        } catch(IOException e) {
+            throw new ElasticsearchParseException("Failed to parse aliases", e);
+        }
+    }
+
+    /**
+     * Adds an alias that will be associated with the index when it gets created
+     */
+    public CreateIndexRequest alias(Alias alias) {
+        this.aliases.add(alias);
+        return this;
+    }
+
+    /**
+     * Adds aliases that will be associated with the index when it gets created
+     */
+    public CreateIndexRequest aliases(Collection<Alias> aliases) {
+        this.aliases.addAll(aliases);
+        return this;
+    }
+
+    /**
+     * Sets the settings and mappings as a single source.
+     *
+     * Note that the mapping definition should *not* be nested under a type name.
+     */
+    public CreateIndexRequest source(String source, XContentType xContentType) {
+        return source(new BytesArray(source), xContentType);
+    }
+
+    /**
+     * Sets the settings and mappings as a single source.
+     *
+     * Note that the mapping definition should *not* be nested under a type name.
+     */
+    public CreateIndexRequest source(XContentBuilder source) {
+        return source(BytesReference.bytes(source), source.contentType());
+    }
+
+    /**
+     * Sets the settings and mappings as a single source.
+     *
+     * Note that the mapping definition should *not* be nested under a type name.
+     */
+    public CreateIndexRequest source(byte[] source, XContentType xContentType) {
+        return source(source, 0, source.length, xContentType);
+    }
+
+    /**
+     * Sets the settings and mappings as a single source.
+     *
+     * Note that the mapping definition should *not* be nested under a type name.
+     */
+    public CreateIndexRequest source(byte[] source, int offset, int length, XContentType xContentType) {
```

Comment:
do we need all these ways to set `source` here?

Reply:
in general, i think we should try to clean up these classes if we can, so that people have less ways to set these values on the client side objects.

Reply:
I'll take a stab at removing some of the less important setters on this class.